mico-core/src/main/java/io/github/ust/mico/core/REST/ServiceController.java

```diff
@@ -71,6 +71,25 @@
         }
     }
 
+    @DeleteMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}")
+    public ResponseEntity deleteService(@PathVariable(PATH_VARIABLE_SHORT_NAME) String shortName,
+                                        @PathVariable(PATH_VARIABLE_VERSION) String version) {
+        Optional<Service> serviceOpt = serviceRepository.findByShortNameAndVersion(shortName, version);
+        if(!serviceOpt.isPresent()){
+            return ResponseEntity.notFound().build();
+        } else {
+            //only allow a deletion if the service has no dependees or depeners
+            Service service = serviceOpt.get();
+            //TODO: && getDependers(shortName,version) == null --> check for dependers missing
+            if(service.getDependsOn() == null){
```

Comment:
Why does the `DependsOn` have to be `null`? The services mentioned in the `DependsOn` might exist and be necessary without the service to delete.